src/messaging/ReliableMessageContext.cpp

```diff
@@ -51,6 +54,22 @@ ReliableMessageMgr * ReliableMessageContext::GetReliableMessageMgr()
     return static_cast<ExchangeContext *>(this)->GetExchangeMgr()->GetReliableMessageMgr();
 }
 
+void ReliableMessageContext::SetMessageNotAcked(bool messageNotAcked)
```

Comment:
This function went from being idempotent to very much not so.  Are you sure all the consumers handle that properly?

Reply:
It is only called the creation and destruction of a reliable message entry. So when a message requiring an ack is sent (entry created) and when the ack is received or the maximum number of retries was attempted. It is possible I missed something. But from what I understood this is handled properly.